commit 7f8c750541790de5a380409466cf07cfe19c3651
Author: Lukas Tinkl <lukas@kde.org>
Date:   Wed Jun 6 15:47:39 2012 +0200

    Revert "don't consider external batteries (mice, keyboard, BT) as primary"
    
    This reverts commit d1661d1638851fd950c5926fb2040ea2abe2a8f0.
    
    This causes more trouble than expected, newer kernels/upower contain a bug that (in my case
    and many others) marks ALL batteries as non-powersupply.
    
    BUG:301033
    CCBUG:300787

diff --git a/solid/solid/backends/upower/upowerbattery.cpp b/solid/solid/backends/upower/upowerbattery.cpp
index 9c5f675..ab555d4 100644
--- a/solid/solid/backends/upower/upowerbattery.cpp
+++ b/solid/solid/backends/upower/upowerbattery.cpp
@@ -44,18 +44,15 @@ Solid::Battery::BatteryType Battery::type() const
 {
     Solid::Battery::BatteryType result = Solid::Battery::UnknownBattery;
     const uint t = m_device->prop("Type").toUInt();
-    bool powerSupply = m_device->prop("PowerSupply").toBool();
     switch (t)
     {
         case 1: // TODO "Line Power"
             break;
         case 2:
-            if (powerSupply)
-                result = Solid::Battery::PrimaryBattery;
+            result = Solid::Battery::PrimaryBattery;
             break;
         case 3:
-            if (powerSupply)
-                result = Solid::Battery::UpsBattery;
+            result = Solid::Battery::UpsBattery;
             break;
         case 4:
             result = Solid::Battery::MonitorBattery;
